### PR TITLE
 Implement _mm_sra_epi16 and _mm_sra_epi32

### PR DESCRIPTION
**Pull request by sebpop**
This depends on PR #29.

### CODE REVIEW

File: sse2neon.h [hunk 1]
```diff
@@ -2990,6 +3005,41 @@ FORCE_INLINE __m128i _mm_loadu_si128(const __m128i *p)
...
+//   r2 := a2 >> count
+//   r3 := a3 >> count
+//
+// https://msdn.microsoft.com/en-us/library/ce40009e(v%3dvs.100).aspx
+FORCE_INLINE __m128i _mm_sra_epi32(__m128i a, __m128i count)
+{
+    int c = (int64_t) vget_low_s64((int64x2_t) count);
+    if (c > 31)
+        return _mm_cmplt_epi32(a, _mm_setzero_si128());
+    return (__m128i) vshlq_s32((int32x4_t) a, vdupq_n_s32(-c));
```
[the remark this answers is not in the document]
Comment (sebpop) [reply sse2neon.h]: done

File: sse2neon.h [hunk 2]
```diff
@@ -2990,6 +3005,41 @@ FORCE_INLINE __m128i _mm_loadu_si128(const __m128i *p)
...
+//   ...
+//   r7 := a7 >> count
+//
+// https://msdn.microsoft.com/en-us/library/3c9997dk(v%3dvs.90).aspx
+FORCE_INLINE __m128i _mm_sra_epi16(__m128i a, __m128i count)
+{
+    int c = (int64_t) vget_low_s64((int64x2_t) count);
+    if (c > 15)
+        return _mm_cmplt_epi16(a, _mm_setzero_si128());
+    return (__m128i) vshlq_s16((int16x8_t) a, vdupq_n_s16(-c));
```
[the remark this answers is not in the document]
Comment (sebpop) [reply sse2neon.h]: done

File: sse2neon.h [hunk 3]
```diff
@@ -2602,6 +2602,21 @@ FORCE_INLINE __m128i _mm_cmpgt_epi8(__m128i a, __m128i b)
...
 
+// Compares the 8 signed 16-bit integers in a and the 8 signed 16-bit integers
+// in b for less than.
+//
+//   r0 := (a0 < b0) ? 0xffff : 0x0
+//   r1 := (a1 < b1) ? 0xffff : 0x0
+//   ...
+//   r7 := (a7 < b7) ? 0xffff : 0x0
+//
+// https://technet.microsoft.com/en-us/library/t863edb2(v=vs.100).aspx
```
Comment (jserv) [inline sse2neon.h:2613]: Did you find any valid URL which starts with `msdn.microsoft.com`?

Comment (sebpop) [reply sse2neon.h:2613]: My understanding is that all pages on technet and msdn are the same as long as the hash key remains the same:
https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2010/t863edb2(v=vs.100)
works as well as https://technet.microsoft.com/en-us/library/t863edb2(v=vs.100).aspx

Comment (jserv) [reply sse2neon.h:2613]: Thanks for clarification. I was thinking of the feasibility to unify all comments with shorter URLs though.

Comment (sebpop) [reply sse2neon.h:2613]: Maybe we can commit the comments change as a separate patch?

Comment (sebpop) [reply sse2neon.h:2613]: Can we commit this PR as is, or do I need to change the comment with a different url?

Comment (jserv) [reply sse2neon.h:2613]: Please ignore my previous question. Let's move forward.

**Pull request merged**